### PR TITLE
Add ContactPage configuration for Meta

### PR DESCRIPTION
**Pull request by AgentIsai**
Creates the 'requestaccount' ContactPage form on Meta

### CODE REVIEW

File: LocalWiki.php [hunk 1]
```diff
@@ -139,6 +139,53 @@ function onBeforePageDisplay( OutputPage $out ) {
...
+	case 'metawiki':
+		$wgContactConfig['requestaccount'] = [
+			'RecipientUser' => 'Miraheze CVT',
+			'SenderName' => 'Account Creation Request Form via Meta',
+			'RequireDetails' => true,
+			'IncludeIP' => false,
+			'MustBeLoggedIn' => false,
+			'AdditionalFields' => [
+				'SelectIssue' => [
+					'label-message' => 'contactpage-requestaccount-selectissue',
```
Comment (Universal-Omega) [inline LocalWiki.php:152]: This and all following messages exist?

File: LocalWiki.php [hunk 2]
```diff
@@ -139,6 +139,53 @@ function onBeforePageDisplay( OutputPage $out ) {
 	case 'loginwiki':
 		wfLoadExtension( 'GlobalWatchlist' );
 
+		break;
+	case 'metawiki':
+		$wgContactConfig['requestaccount'] = [
+			'RecipientUser' => 'Miraheze CVT',
```
Comment (Universal-Omega) [inline LocalWiki.php:145]: This user exists?

**Review by Universal-Omega — COMMENTED** [review]
Additionally, looking at source, it appears ContactPage itself requires a captcha as well.

Default fields still exist also, https://github.com/wikimedia/mediawiki-extensions-ContactPage/blob/33f93e31afaa0375697b1229335caf684cdc63b2/includes/SpecialContact.php#L172-L198, just an FYI I thought I would mention.

File: LocalWiki.php (hunk 2 above)
Comment (AgentIsai) [reply LocalWiki.php:145]: It does

File: LocalWiki.php (hunk 1 above)
Comment (AgentIsai) [reply LocalWiki.php:152]: They will be created on Meta once this is merged

**Comment by AgentIsai** [comment]
> Additionally, looking at source, it appears ContactPage itself requires a captcha as well.

It does not actually. I have tested this in a local enviornment and did so without having Captcha configured for it. While you can configure Captcha, it is not required.

> Default fields still exist also, https://github.com/wikimedia/mediawiki-extensions-ContactPage/blob/33f93e31afaa0375697b1229335caf684cdc63b2/includes/SpecialContact.php#L172-L198, just an FYI I thought I would mention.

Indeed, thanks for mentioning it

**Comment by Universal-Omega** [comment]
Also, it may be overridden by https://github.com/miraheze/mw-config/blob/5eadef96a72d7687cba5ccbe593e013d49d02a9e/GlobalSettings.php#L246 since the extension merges new form type with the defaukt config. Not sure which one overrides.

**Comment by AgentIsai** [comment]
GlobalSettings seems to be loaded before LocalWiki, perhaps this will override that?

**Comment by Universal-Omega** [comment]
And it only loads non-default forms when using URL query (?formtype) or subpage, Special:ContactPage/formtype, something else I thought I would mention, not by default. https://github.com/wikimedia/mediawiki-extensions-ContactPage/blob/33f93e31afaa0375697b1229335caf684cdc63b2/includes/SpecialContact.php#L90

**Comment by Universal-Omega** [comment]
> GlobalSettings seems to be loaded before LocalWiki, perhaps this will override that?

If you set the config fully, this is appending to the config, not setting it.

**Comment by Universal-Omega** [comment]
And the default key is also **required** by the extension.

**Comment by AgentIsai** [comment]
Is it? Wikimedia projects don't configure it, see [Meta-Wiki](https://meta.wikimedia.org/wiki/Special:Contact)

**Comment by Universal-Omega** [comment]
> Is it? Wikimedia projects don't configure it, see [Meta-Wiki](https://meta.wikimedia.org/wiki/Special:Contact)

Yes they do. https://github.com/wikimedia/operations-mediawiki-config/blob/59219fff5298f34cf0df34828f99e0e3dbef054f/wmf-config/CommonSettings.php#L1439-L1456

**Pull request merged**